(Opensearch): AuthorizationException thrown in opensearch_index.py when first deploy

### Describe the bug

`AuthorizationException` occurs in `opensearch_index.py` running as custom resource and resource creation fails.

### Expected Behavior

`AuthorizationException` should not occur.

### Current Behavior

`AuthorizationException` should not occurs.

Full error log in CloudWatch logs

```

[WARNING]	2024-03-14T02:06:29.469Z	e9bb4f1c-04c8-47c9-b64c-bdd1862ba5ef	PUT https://yvbe8pdu86uryqmsumxg.us-west-2.aoss.amazonaws.com:443/bedrock-knowledge-base-default-index?wait_for_active_shards=all [status:403 request:0.073s]
--
[ERROR]	2024-03-14T02:06:29.470Z	e9bb4f1c-04c8-47c9-b64c-bdd1862ba5ef	Error creating index bedrock-knowledge-base-default-index
[ERROR]	2024-03-14T02:06:29.470Z	e9bb4f1c-04c8-47c9-b64c-bdd1862ba5ef	AuthorizationException(403, 'security_exception', 'OpenSearch exception [type=authorization_exception, reason=User does not have permissions for the requested resource]')Traceback (most recent call last):  File "/var/task/custom_resources/opensearch_index.py", line 177, in handle_create    create_index(client, index_name, mapping)  File "/var/task/custom_resources/opensearch_index.py", line 143, in create_index    client.indices.create(  File "/var/task/opensearchpy/client/utils.py", line 181, in _wrapped    return func(*args, params=params, headers=headers, **kwargs)           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  File "/var/task/opensearchpy/client/indices.py", line 166, in create    return self.transport.perform_request(           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  File "/var/task/opensearchpy/transport.py", line 448, in perform_request    raise e  File "/var/task/opensearchpy/transport.py", line 409, in perform_request    status, headers_response, data = connection.perform_request(                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^  File "/var/task/opensearchpy/connection/http_requests.py", line 232, in perform_request    self._raise_error(  File "/var/task/opensearchpy/connection/base.py", line 316, in _raise_error    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(opensearchpy.exceptions.AuthorizationException: AuthorizationException(403, 'security_exception', 'OpenSearch exception [type=authorization_exception, reason=User does not have permissions for the requested resource]')
[WARNING]	2024-03-14T02:06:30.640Z	e9bb4f1c-04c8-47c9-b64c-bdd1862ba5ef	PUT https://yvbe8pdu86uryqmsumxg.us-west-2.aoss.amazonaws.com:443/bedrock-knowledge-base-default-index?wait_for_active_shards=all [status:403 request:0.090s]
[ERROR]	2024-03-14T02:06:30.640Z	e9bb4f1c-04c8-47c9-b64c-bdd1862ba5ef	Error creating index bedrock-knowledge-base-default-index
[ERROR]	2024-03-14T02:06:30.640Z	e9bb4f1c-04c8-47c9-b64c-bdd1862ba5ef	AuthorizationException(403, 'security_exception', 'OpenSearch exception [type=authorization_exception, reason=User does not have permissions for the requested resource]')Traceback (most recent call last):  File "/var/task/custom_resources/opensearch_index.py", line 177, in handle_create    create_index(client, index_name, mapping)  File "/var/task/custom_resources/opensearch_index.py", line 143, in create_index    client.indices.create(  File "/var/task/opensearchpy/client/utils.py", line 181, in _wrapped    return func(*args, params=params, headers=headers, **kwargs)           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  File "/var/task/opensearchpy/client/indices.py", line 166, in create    return self.transport.perform_request(           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  File "/var/task/opensearchpy/transport.py", line 448, in perform_request    raise e  File "/var/task/opensearchpy/transport.py", line 409, in perform_request    status, headers_response, data = connection.perform_request(                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^  File "/var/task/opensearchpy/connection/http_requests.py", line 232, in perform_request    self._raise_error(  File "/var/task/opensearchpy/connection/base.py", line 316, in _raise_error    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(opensearchpy.exceptions.AuthorizationException: AuthorizationException(403, 'security_exception', 'OpenSearch exception [type=authorization_exception, reason=User does not have permissions for the requested resource]')
[WARNING]	2024-03-14T02:06:32.883Z	e9bb4f1c-04c8-47c9-b64c-bdd1862ba5ef	PUT https://yvbe8pdu86uryqmsumxg.us-west-2.aoss.amazonaws.com:443/bedrock-knowledge-base-default-index?wait_for_active_shards=all [status:403 request:0.036s]
[ERROR]	2024-03-14T02:06:32.884Z	e9bb4f1c-04c8-47c9-b64c-bdd1862ba5ef	Error creating index bedrock-knowledge-base-default-index
[ERROR]	2024-03-14T02:06:32.884Z	e9bb4f1c-04c8-47c9-b64c-bdd1862ba5ef	AuthorizationException(403, 'security_exception', 'OpenSearch exception [type=authorization_exception, reason=User does not have permissions for the requested resource]')Traceback (most recent call last):  File "/var/task/custom_resources/opensearch_index.py", line 177, in handle_create    create_index(client, index_name, mapping)  File "/var/task/custom_resources/opensearch_index.py", line 143, in create_index    client.indices.create(  File "/var/task/opensearchpy/client/utils.py", line 181, in _wrapped    return func(*args, params=params, headers=headers, **kwargs)           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  File "/var/task/opensearchpy/client/indices.py", line 166, in create    return self.transport.perform_request(           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  File "/var/task/opensearchpy/transport.py", line 448, in perform_request    raise e  File "/var/task/opensearchpy/transport.py", line 409, in perform_request    status, headers_response, data = connection.perform_request(                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^  File "/var/task/opensearchpy/connection/http_requests.py", line 232, in perform_request    self._raise_error(  File "/var/task/opensearchpy/connection/base.py", line 316, in _raise_error    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(opensearchpy.exceptions.AuthorizationException: AuthorizationException(403, 'security_exception', 'OpenSearch exception [type=authorization_exception, reason=User does not have permissions for the requested resource]')
LAMBDA_WARNING: Unhandled exception. The most likely cause is an issue in the function code. However, in rare cases, a Lambda runtime update can cause unexpected function behavior. For functions using managed runtimes, runtime updates can be triggered by a function change, or can be applied automatically. To determine if the runtime has been updated, check the runtime version in the INIT_START log entry. If this error correlates with a change in the runtime version, you may be able to mitigate this error by temporarily rolling back to the previous runtime version. For more information, see https://docs.aws.amazon.com/lambda/latest/dg/runtimes-update.html
[ERROR] RetryError: RetryError[<Future at 0x7f2b6c53d2b0 state=finished raised AuthorizationException>]Traceback (most recent call last):  File "/var/task/custom_resources/__init__.py", line 34, in on_event    return on_event_opensearch_index(event, context)  File "/var/task/custom_resources/opensearch_index.py", line 238, in on_event    return on_create(event)  File "/var/task/custom_resources/opensearch_index.py", line 205, in on_create    physical_id = handle_create(  File "/var/task/tenacity/__init__.py", line 289, in wrapped_f    return self(f, *args, **kw)  File "/var/task/tenacity/__init__.py", line 379, in __call__    do = self.iter(retry_state=retry_state)  File "/var/task/tenacity/__init__.py", line 326, in iter    raise retry_exc from fut.exception()


```

### Reproduction Steps

This problem occurs probabilistically when attempting to deploy to a specific AWS account. Running the same code is unlikely to reproduce the problem.

Nevertheless, here is a snippet of the code that was executed just in case.

```
import { App } from 'sst/constructs/App';
import { Stack } from 'sst/constructs/Stack';
import { bedrock } from '@cdklabs/generative-ai-cdk-constructs';
import { S3Bucket } from '../../prototyping/constructs/S3Bucket';
import { Bucket, IBucket } from 'aws-cdk-lib/aws-s3';
import { BedrockS3DataSource, RagBedrockConfig } from '../../types';

type BedrockProps = {
  stack: Stack;
  app: App;
  ragBedrockConfig: RagBedrockConfig;
  allowedOrigins: any;
};

export function createKnowledgeBase({
  stack,
  app,
  ragBedrockConfig,
  allowedOrigins,
}: BedrockProps) {
  const knowledgeBase = new bedrock.KnowledgeBase(stack, 'KnowledgeBase', {
    embeddingsModel: bedrock.BedrockFoundationModel.TITAN_EMBED_TEXT_V1,
    description: ragBedrockConfig.description,
  });

  const s3DataSources: Array<BedrockS3DataSource> = [];

  ragBedrockConfig.s3dataSourceConfig.forEach((dataSource) => {
    let bucket: Bucket | IBucket;
    let bucketArn = dataSource.bucketArn;

    if (bucketArn) {
      bucket = Bucket.fromBucketArn(
        stack,
        `KnowledgeBase-S3-${dataSource.dataSourceName}`,
        bucketArn,
      );
    } else {
      bucket = new S3Bucket(stack, `KnowledgeBase-S3-${dataSource.dataSourceName}`, {
        sstApp: app,
        allowedOrigins,
      }).bucket;
    }

    new bedrock.S3DataSource(stack, `KnowledgeBase-DS-${dataSource.dataSourceName}`, {
      bucket,
      inclusionPrefixes: dataSource.inclusionPrefixes,
      kmsKey: dataSource.kmsKey,
      knowledgeBase,
      dataSourceName: dataSource.dataSourceName,
      chunkingStrategy: bedrock.ChunkingStrategy.FIXED_SIZE,
      maxTokens: 500,
      overlapPercentage: 20,
    });

    s3DataSources.push({
      bucket,
      inclusionPrefixes: dataSource.inclusionPrefixes ?? [],
    });
  });
  return { knowledgeBase, s3DataSources };
}

```

### Possible Solution

I solved this editting the code of `opensearch_index.py` in `node_modlues` like:

``` python3
# Add retry on AuthorizationException to mitigate policy creation race condition
@retry(
    retry=retry_if_exception_type(AuthorizationException),
    stop=stop_after_attempt(200),
    wait=wait_exponential_jitter(1, 3),
)
```

After this fix, I don't see the failure.

### Additional Information/Context

_No response_

### CDK CLI Version

2.124.0

### Framework Version

_No response_

### Node.js Version

v18.18.0

### OS

Mac OS sonoma

### Language

Typescript

### Language Version

_No response_

### Region experiencing the issue

us-west-2

### Code modification

Yes, the change is very small and in **possible solution**.  If needed, I'll push to my repo.

### Other information

_No response_

### Service quota

- [X] I have reviewed the service quotas for this construct

--- a/lambda/opensearch-serverless-custom-resources/custom_resources/opensearch_index.py
+++ b/lambda/opensearch-serverless-custom-resources/custom_resources/opensearch_index.py
@@ -159,7 +159,7 @@ def create_index(client: OpenSearch, index_name: str, mapping: dict[str, str]) -
 # Add retry on AuthorizationException to mitigate policy creation race condition
 @retry(
     retry=retry_if_exception_type(AuthorizationException),
-    stop=stop_after_attempt(3),
+    stop=stop_after_attempt(10),
     wait=wait_exponential_jitter(1, 3),
 )
 def handle_create(

